tests/posix/wcwidth.c

```diff
@@ -0,0 +1,56 @@
+#include <assert.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <wchar.h>
+
+/*
+ * The code in this test is taken from https://github.com/termux/wcwidth/,
+ * under the following license:
+ *
+ * Copyright (C) Fredrik Fornwall 2016.
+ * Distributed under the MIT License.
+ *
+ * Implementation of wcwidth(3) as a C port of:
+ * https://github.com/jquast/wcwidth
+ *
+ * Report issues at:
+ * https://github.com/termux/wcwidth
+ */
+
+static int tests_run;
+static int test_failures;
+
+void assertWidthIs(int expected_width, wchar_t c) {
+	tests_run++;
+	int actual_width = wcwidth(c);
+	if (actual_width != expected_width) {
+		fprintf(stderr, "ERROR: wcwidth(U+%04x) returned %d, expected %d\n", c, actual_width, expected_width);
+		test_failures++;
+	}
+}
+
+int main() {
+	assertWidthIs(1, 'a');
```

Comment:
Does this succeed on host by setting locale before the line above?

Reply:
Setting locale makes it work, other than one test excluded from host testing.